packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -366,17 +366,29 @@ const VirtualListBaseFactory = (type) => {
 			if (isPrimaryDirectionVertical) {
 				if (isUpKey && row) {
 					targetIndex = index - dimensionToExtent;
-				} else if (isDownKey && isNextRow) {
-					targetIndex = index + dimensionToExtent;
+				} else if (isDownKey) {
+					if (isNextRow) {
+						targetIndex = index + dimensionToExtent;
+					} else if ( // if the next line is the last line and there is no item below the current item
+						Math.floor((index + dimensionToExtent ) / dimensionToExtent) === Math.floor((dataSize - 1) / dimensionToExtent)
```

Comment:
@ybsung  What about this?
`targetIndex = Math.min(index + dimensionToExtent, dataSize - 1)`
This has a little different behavior from the PRed code and it may be sufficient to work if the behavior is acceptable.